pkg/docker/docker.go

```diff
@@ -491,9 +488,18 @@ func (d *stiDocker) CheckAndPullImage(name string) (*api.Image, error) {
 	return image, nil
 }
 
+// ImageShortName shorten image name when tag longer than 18 characters.
```

Comment:
feels like there would already be a k8s or docker utility function around this but my quick searches could not find one

Where did the 18 limit come from ?  Why not 20 or 12?

Reply:
That was already present in s2i, I only moved that to a common place. I'm not sure why this limitation is in place, but in the case of `buildah` it should produce a similar user experience, hence the reuse.

Reply:
Maybe we would like to open a new issue with this subject, track it separately.